### ISSUE TITLE
Consider using class factory to create concrete classes of UQ test functions

### ISSUE THREAD

**Issue by damar-wicaksono** [damar-wicaksono/uqtestfuns#212]
Each class of UQ test functions is a concrete implementation of `UQTestFunABC` abstract base class.
Currently, the concrete implementation is created in a separate module that contains the class definition.
There is a lot of boilerplate codes.

Inspired by this [blog post](https://www.alexkorablev.com/cast-function.html), consider having a factory class function that dynamically create the concrete class upon package import.  The concrete class would still be located in a separate module because each module contains the relevant data but the concrete class itself is created by a function call.

For instance, to create a concrete implementation of the one-dimensional two-regime Gramacy (2007) function, we can type:

```python
Gramacy2007 = create_concrete_uqtestfun(
   "Gramacy2007",
   evaluate,  # Already defined
   # other important attributes
)
```

This way, we can reduce a lot of boilerplate codes in the codebase.

While there should not be major changes to the high-level functionalities of the package, this issue constitutes quite a major refactoring of the codebase.

**Comment by damar-wicaksono** [comment]
From PR #218, the abstract base class itself has been refactored to minimize the amount of boilerplate codes even without using factory class. All built-in UQ test functions, be they fixed dimension or variable dimension, share the same interface and derived from `UQTestFunABC`.  `spatial_dimension` is a parameter in the constructor for both. However, only when the test functions are of variable dimension, the parameter becomes active.

**Comment by damar-wicaksono** [comment]
This issue has been resolved by PR #218.
